ui/src/components/Rules/Rule/RuleCreator/index.tsx

```diff
@@ -123,6 +123,16 @@ const RuleCreator = (props: iRuleCreator) => {
       let rules = rulesCreated.current?.filter((el) => el.id !== id)
       rules = rules.map((e) => {
         e.id = e.id > id ? e.id - 1 : e.id
+
+        if (
+          section === 1 &&
+          rules.filter((r) => r.rule.section === 0).length === 0
+        ) {
+          if (e.rule.section && e.rule.section > 0) {
+            e.rule.section -= 1
+          }
+        }
```

Comment:
You could move the filter check outside of the loop as it's always the same. You also don't need the `> 0` check as `0 == false`.

```suggestion
        const section1IsEmpty = !rules.some((r) => r.rule.section === 0)
        // ... rules.map
        if (
          section1IsEmpty &&
          section === 1 &&
          e.rule.section
        ) {
            e.rule.section -= 1
        }
```

Reply:
Thank you for the tips! I've fixed them.